src/components/datetime/datetime.ts

```diff
@@ -619,28 +619,14 @@ export class DateTime extends Ion implements AfterContentInit, ControlValueAcces
       }
     }
 
-    // default to assuming this month has 31 days
```

Comment:
@flivni yes please! let's keep this PR as simple as possible, can you revert this changes? so it is easier to track future problems.

We can spend time later refactoring the code to look nicer :)